ui/pages/permissions-connect/snap-install/snap-install.js

```diff
@@ -73,21 +73,28 @@ export default function SnapInstall({
       </div>
       <div className="footers">
         {targetSubjectMetadata.sourceCode ? (
-          <div className="source-code">
-            <div className="text">{t('areYouDeveloper')}</div>
-            <div
-              className="link"
-              onClick={() =>
-                global.platform.openTab({
-                  url: targetSubjectMetadata.sourceCode,
-                })
-              }
-            >
-              {t('openSourceCode')}
+          <>
+            <div className="source-code">
+              <div className="text">{t('areYouDeveloper')}</div>
+              <div
+                className="link"
+                onClick={() =>
+                  global.platform.openTab({
+                    url: targetSubjectMetadata.sourceCode,
+                  })
+                }
+              >
+                {t('openSourceCode')}
+              </div>
             </div>
+            <PermissionsConnectFooter />
+          </>
+        ) : (
+          <div className="snap-install__footer--no-source-code">
```

Comment:
Why was this div needed? I don't see anything in the document about this, and it seems incongruous to have a wrapper div for the "no source" case, but not for the "source" case.

Reply:
I added the wrapper div to introduce some padding on the top of `PermissionsConnectFooter` in the case of no source code, it's not needed when there is source code.

Reply:
Ok. I think this would be far simpler to manage in CSS, by setting the padding/margins of the `footers` and `headers` elements, rather than conditionally adding wrapper divs. But this looks like a lot to untangle right now - we can clean this up later. The styles of this component are very unconventional at the moment.